controllers/argocd_metrics_controller.go

```diff
@@ -274,6 +282,42 @@ func (r *ArgoCDMetricsReconciler) createServiceMonitorIfAbsent(namespace string,
 	return err
 }
 
+func (r *ArgoCDMetricsReconciler) reconcileOperatorMetricsServiceMonitor(reqLogger logr.Logger) error {
+
+	data, err := os.ReadFile(operatorPodNamespacePath)
+	if err != nil {
+		reqLogger.Error(err, "Error retrieving operator's running namespace")
+		return err
+	}
+
+	operatorNS := string(data)
+	desiredMetricsServerName := operatorMetricsServiceName + "." + operatorNS + ".svc"
+
+	existingServiceMonitor := &monitoringv1.ServiceMonitor{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: operatorMetricsMonitorName, Namespace: operatorNS}, existingServiceMonitor)
+
+	if err == nil {
+		changed := false
+
+		if existingServiceMonitor.Spec.Endpoints[0].TLSConfig.ServerName != desiredMetricsServerName {
+			existingServiceMonitor.Spec.Endpoints[0].TLSConfig.ServerName = desiredMetricsServerName
+			changed = true
+		}
+
+		if changed {
+			return r.Client.Update(context.TODO(), existingServiceMonitor)
+		}
+	} else if err != nil {
```

Comment:
This code can simplified with `if..else`. If err, return `err` early. Else, check for desired server name and update it.

Reply:
updated, thanks